packages/react/src/ProgressBar/ProgressBar.tsx

```diff
@@ -60,37 +57,76 @@ const ProgressContainer = styled.span<StyledProgressContainerProps>`
   ${sx};
 `
 
+export type ProgressBarItems = React.HTMLAttributes<HTMLSpanElement> & {'aria-label'?: string} & ProgressProp & SxProp
+
+export const Item = forwardRef<HTMLSpanElement, ProgressBarItems>(
+  (
+    {progress, 'aria-label': ariaLabel, 'aria-valuenow': ariaValueNow, 'aria-valuetext': ariaValueText, ...rest},
+    forwardRef,
+  ) => {
+    const progressAsNumber = typeof progress === 'string' ? parseInt(progress, 10) : progress
+
+    const ariaAttributes = {
+      'aria-valuenow':
+        ariaValueNow || (progressAsNumber !== undefined && progressAsNumber >= 0 ? Math.round(progressAsNumber) : 0),
+      'aria-valuemin': 0,
+      'aria-valuemax': 100,
+      'aria-valuetext': ariaValueText,
+    }
+
+    return (
+      <ProgressItem
+        role="progressbar"
+        aria-label={ariaLabel}
+        ref={forwardRef}
+        progress={progress}
+        {...ariaAttributes}
+        {...rest}
+      />
+    )
+  },
+)
+
+Item.displayName = 'ProgressBar.Item'
+
 export type ProgressBarProps = React.HTMLAttributes<HTMLSpanElement> & {bg?: string} & StyledProgressContainerProps &
   ProgressProp
 
 export const ProgressBar = forwardRef<HTMLSpanElement, ProgressBarProps>(
   (
-    {animated, progress, bg = 'success.emphasis', barSize = 'default', children, ...rest}: ProgressBarProps,
+    {
+      animated,
+      progress,
+      bg = 'success.emphasis',
+      barSize = 'default',
+      children,
+      'aria-label': ariaLabel,
+      'aria-valuenow': ariaValueNow,
+      'aria-valuetext': ariaValueText,
+      ...rest
+    }: ProgressBarProps,
     forwardRef,
   ) => {
     if (children && progress) {
       throw new Error('You should pass `progress` or children, not both.')
     }
 
-    warning(
-      children &&
-        typeof (rest as React.AriaAttributes)['aria-valuenow'] === 'undefined' &&
-        typeof (rest as React.AriaAttributes)['aria-valuetext'] === 'undefined',
-      'Expected `aria-valuenow` or `aria-valuetext` to be provided to <ProgressBar>. Provide one of these values so screen reader users can determine the current progress. This warning will become an error in the next major release.',
-    )
-
-    const progressAsNumber = typeof progress === 'string' ? parseInt(progress, 10) : progress
-
-    const ariaAttributes = {
-      'aria-valuenow': progressAsNumber ? Math.round(progressAsNumber) : undefined,
-      'aria-valuemin': 0,
-      'aria-valuemax': 100,
-      'aria-valuetext': progressAsNumber ? `${Math.round(progressAsNumber)}%` : undefined,
-    }
+    const validChildren = React.Children.toArray(children).length
 
     return (
-      <ProgressContainer ref={forwardRef} role="progressbar" barSize={barSize} {...ariaAttributes} {...rest}>
-        {children ?? <Item data-animated={animated} progress={progress} bg={bg} />}
+      <ProgressContainer ref={forwardRef} barSize={barSize} {...rest}>
+        {validChildren ? (
+          children
```

Comment:
For this check, would it be enough to check for the presence of children and toggle or is the count important / used somewhere else?

Reply:
Right now, you can use the `ProgressBar` alone, or with children. If you use it alone, you can apply all the props directly to `<ProgressBar>`, but with children, you need to apply them to the individual items. 

I think I understand the ask, are you asking if we can switch up the current conditional? 🤔